### PR TITLE
Feat 347 test matrix

### PR DESCRIPTION
**Pull request by rtitmuss**

### CODE REVIEW

File: src/components/configure/keymap/Keymap.tsx [hunk 1]
```diff
@@ -169,6 +171,21 @@ export default class Keymap extends React.Component<
     }
   }
 
+  componentDidUpdate(prevProps: KeymapPropsType, prevState: OwnKeymapStateType) {
+    if (prevProps.testMatrix !== this.props.testMatrix) {
+      clearInterval(this.state.interval);
+      if (this.props.testMatrix) {
+        this.setState({ interval: setInterval(() => {
+            this.props.fetchSwitchMatrixState!();
+          }, 100) });
```
Comment (yoichiro) [inline src/components/configure/keymap/Keymap.tsx:180]: @rtitmuss When testing this code on my Linux, it worked normally and the CPU load was not almost increased. What OS were you using during developing? We need to test this feature on major OSs, I think.

File: src/actions/hid.action.ts [hunk 2]
```diff
@@ -356,6 +356,39 @@ export const hidActionsThunk = {
...
+    let i = 0;
+    for (let row = 0; row < rows; row++) {
+      let rowState = 0;
+      for (let colIdx = 0; colIdx < Math.ceil(cols/8); colIdx++) {
+        rowState = (rowState << 8) | state[i++];
+      }
+
+      for (let col = 0; col < cols; col++) {
+        if ((rowState & (1 << col)) !== 0) {
+          dispatch(AppActions.addTestedMatrix(row + ',' + col));
```
Comment (yoichiro) [inline src/actions/hid.action.ts:387]: @rtitmuss To handle key release events and to clear highlighting of each key top, you need to call the interface to clear them. @adamrocker will create the new interface. So could you call it after implementing of the new interface?

**Review by yoichiro — CHANGES_REQUESTED** [review]
@rtitmuss I posted some comments. Could you confirm them? Thank you!

**Comment by yoichiro** [comment]
@rtitmuss BTW, the reason why the CI was failed is about the code format. We're using a prettier. Please execute `yarn format` before pushing your code. I intend to write about this on README.md.

**Comment by rtitmuss** [comment]
> @rtitmuss BTW, the reason why the CI was failed is about the code format. We're using a prettier. Please execute `yarn format` before pushing your code. I intend to write about this on README.md.

Thanks for the top, I was fixing these by hand :)

File: src/actions/hid.action.ts (hunk 2 above)
Comment (rtitmuss) [reply src/actions/hid.action.ts:387]: Yes, I'll make the changes to support the new interface.

File: src/components/configure/keymap/Keymap.tsx (hunk 1 above)
Comment (rtitmuss) [reply src/components/configure/keymap/Keymap.tsx]: I am developing in Chrome on OS X

Comment (yoichiro) [reply src/components/configure/keymap/Keymap.tsx]: Ok, I will test this feature on my Windows 10 (Chrome and Edge) and will report the result.

File: src/actions/hid.action.ts (hunk 2 above)
Comment (yoichiro) [reply src/actions/hid.action.ts:387]: @rtitmuss adamrocker created the new interface at: https://github.com/remap-keys/remap/blob/feat-347-test-matrix/src/components/configure/keymapToolbar/KeymapToolbar.container.ts#L41

You can set the `updateCurrentTestMatrix()` function with an argument. The argument is an array of "Pressed down key position at that time". The `addTestedMatrix()` function is used to highlight the key top. In the other hand, the `updateCurrentTestMatrix()` function is used for representing key press/release state. Could you use the `updateCurrentTestMatrix()` function?

Comment (rtitmuss) [reply src/actions/hid.action.ts]: I've subumitted an update.

Comment (yoichiro) [reply src/actions/hid.action.ts]: Ok, I will test it.

**Comment by yoichiro** [comment]
@rtitmuss Sorry, I heard about key event handling from @adamrocker, the handling code has already been implemented at: https://github.com/remap-keys/remap/blob/00427a63cf5780ceea2b43c983ffae15693208d3/src/components/configure/Configure.tsx#L126-L137

So, I confirmed that the `Tab` key press event is prevented. But, other some keys and key bindings including the `GUI (that is, Win and Command)` can't be prevented because they are handled by the browser and/or OS, I guess. I think that we need to allow some behaviors by pressing them.

**Review by yoichiro — APPROVED** [review]
@rtitmuss LGTM! We move forward to release this feature. Thank you for your contribution!

**Pull request merged**